napari/utils/interactions.py

```diff
@@ -328,11 +328,12 @@ def parse_platform(text: str) -> str:
             text.replace('+', '')
             text.replace('Plus', '+')
         for k, v in KEY_SYMBOLS.items():
+            new_text = f'{k}-' if sys.platform == 'darwin' else k
+            # case where we are replacing the last symbol in the shortcut
+            # (no need to add `-` even when we are on macOS)
             if text.endswith(v):
-                text = text.replace(v, k)
-                assert v not in text
-            else:
-                text = text.replace(v, k + '-')
+                new_text = k
+            text = text.replace(v, new_text)
```

Comment:
And if I understand this fix then your above if platform should not be necessary.

If not can you explain why ?

Reply:
From what I checked seemed like on Windows/Linux the `+` is used to separete things while on mac seemed like the `-` was working (although in the interface you don't see any separator symbol between keys). Anyhow, seems like something else is going on with this then 🤔 Will try to better understand what is causing this :+1:

Reply:
I think there are 3 things that are strings: `QtShortcut`, `PlatformShortcut` and `CanonicalShortcut` , maybe at some point we should/could use python's NewType to differentiate, and that may help.

I think it's a bit messy because `Shortcut` also implement a `__str__`, I think it shouldn't.

Reply:
After checking a little bit more carefully the logic, I think I found it! Seems like the actual issue was a couple of lines above where the string replacement was not being reassigned to the `text` variable